.changeset/unlucky-beans-obey.md

```diff
@@ -0,0 +1,5 @@
+---
+'openzeppelin-solidity': patch
+---
+
+`ERC2771Context`: Return the forwarder address whenever `msg.data.length` is not appended with the request signer address (i.e. length is less than 20 bytes).
```

Comment:
The current description is a bit misleading because it sounds like we can know whether the request signer address is appended, but the only thing we can know is if the length is greater than 20 bytes.

```suggestion
`ERC2771Context`: Return the forwarder address whenever `msg.data` is smaller than 20 bytes as specified by ERC-2771, since the request signer address can't be extracted.
```

Reply:
Reviewing the new changeset...

Reply:
Note that the suggestion you're making is missing that the change only applies when called via the forwarder. Now that I think about it, it seems that the forwarder address is always returned if the calldata length is less than 20 bytes.